routing_common/transit_types.hpp

```diff
@@ -87,34 +87,54 @@ class Stop
   // and deserialization.
 };
 
+class SingleMwmSegment
+{
+public:
+  SingleMwmSegment() = default;
+  explicit SingleMwmSegment(FeatureId featureId, uint32_t segmentIdx, bool forward);
+
+  FeatureId GetFeatureId() const { return m_featureId; }
+  uint32_t GetSegmentIdx() const { return m_segmentIdx; }
+  bool GetForward() const { return m_forward; }
+
+  DECLARE_VISITOR_AND_DEBUG_PRINT(SingleMwmSegment, visitor(m_featureId, "feature_id"),
```

Comment:
Why not "featureId", "segmentIdx"? I mean, why do you prefer unix_hacker_style to camelCase?

Reply:
If your only point is that unix_hacker_style is more common in json, I suggest you to modify json visitor to convert camelCase names to unix_hacker_style names instead.

Reply:
The thing is most part of such names are used to parse json. And according to https://confluence.mail.ru/display/MAPSME/Subway+Data+Preparation json field uses this style. SingleMwmSegment field names are not used for json parsing. But for uniformity I'd prefer to leave such names here.

Reply:
> If your only point is that unix_hacker_style is more common in json, I suggest you to modify json visitor to convert camelCase names to unix_hacker_style names instead.

If so, probably it'll be better to modify the script which generates the json. And modify all the fields name in transit_types.hpp. In that case we have less code. But why you don't like the current naming of json fields and corresponding names in visitors?

Reply:
Because the format of string names differ from the names of the corresponding fields. In general, I'd like the approach to generate names automatically, by omitting "m_" prefix from the fields names.

Reply:
It's possible, but may lead to subtle bugs, when someone will forget to name fields with "m_" prefix, and to unnecessary overhead for each call.

Reply:
Note, that you may DECLARE_VISITOR in the private section, and declare particular json visitors as friends of the class. So, you'll solve the problem.

Reply:
> Note, that you may DECLARE_VISITOR in the private section, and declare particular json visitors as friends of the class. So, you'll solve the problem.

Let's do so.

Reply:
Done